adcm_aio_client/core/objects/cm.py

```diff
@@ -480,15 +486,27 @@ def object(self: Self) -> Object:
     def action(self: Self) -> Action:
         return self._construct(what=Action, from_data=self._data["action"])
 
-    async def wait(self: Self, status_predicate: Callable[[], bool], timeout: int = 30, poll: int = 5) -> None:
-        if self._data["status"] not in (JobStatus.RUNNING, JobStatus.CREATED):
-            return
+    async def wait(
+        self: Self, timeout: int | None, poll_interval: int = 10, exit_condition: Optional[Callable[[], bool]] = None
+    ) -> Self:
+        if self._data["status"] not in ("running", "created"):
+            return self
+
+        if exit_condition is None:
+            exit_condition = self._default_exit_condition
+
+        max_iterations = float("inf") if timeout is None else (timeout // poll_interval)
 
-        for _ in range(timeout // poll):
-            await asyncio.sleep(poll)
-            if status_predicate():
+        for _ in range(int(max_iterations)):
+            await asyncio.sleep(poll_interval)
+            if exit_condition():
                 self._data["status"] = self.get_status()
-                return
+                return self
```

Comment:
IMO exit condition should be checked before `asyncio.sleep` OR at least once before looping

---

IMO `exit_condition` isn't bound to status, so most likely we should update the whole data with refresh

@a-alferov @kuhella What do you think?

I see that we can:
1. Refresh the internal data AND point out that `exit_condition` should refer to this data (it'll be a "side effect" of wait)
2. Don't refresh data at all, then `exit_condition` should consider making all required inquiries (it's strange, but possible)
3. Leave it as it, but then power of `exit_condition` is not at max

Reply:
Solution: 2, not use `self.refresh()`, not update `_data`